src/cell_ref.rs

```diff
@@ -38,19 +34,22 @@ where
     ///     Reaching `usize::MAX` may be possible with
     ///     `std::mem::forget(CellRef::clone(&r))`.
     pub fn try_clone(&self) -> Result<Self, RefOverflow> {
-        self.flag
-            .fetch_update(Ordering::SeqCst, Ordering::SeqCst, |current_value| {
-                if current_value <= REF_LIMIT_MAX {
-                    Some(current_value + 1)
-                } else {
-                    None
-                }
-            })
-            .map(|_| CellRef {
+        let previous_value = self.flag.fetch_add(1, Ordering::Relaxed);
+
+        #[cfg(not(feature = "nightly"))]
+        let overflow = previous_value == usize::MAX;
```

Comment:
The limit should be `isize::MAX as usize`.

And code can assume that other threads won't be able to do `isize::MAX +1` more increments to overflow the counter before decrements would happen.